src/v2i-hub/PortDrayagePlugin/src/WebServiceClient.cpp

```diff
@@ -0,0 +1,277 @@
+#include "WebServiceClient.h"
+
+
+WebServiceClient::WebServiceClient() {
+    // Set polling frequency to 5s
+    polling_frequency = 5;
+    // Initialize API
+    api = new OAIDefaultApi(0);
+    // Setup Server config
+    api->setNewServerForAllOperations(
+        QUrl("http://127.0.0.1:8090"),
+        QString::fromStdString("Unsecured hosting for development"),
+        QMap<QString, OAIServerVariable>()
+    );
+}
+
+WebServiceClient::WebServiceClient(std::string host, uint8_t port, bool secure , uint16_t polling_frequency ) {
+    this->polling_frequency = polling_frequency;
+    // OAIDefaultApi api;
+
+    // Create URL
+    QUrl url;
+    url.setHost(QString::fromStdString(host));
+    url.setPort(port);
+    if ( secure ) {
+        url.setScheme(QString::fromStdString("https"));
+    }
+    else {
+        url.setScheme(QString::fromStdString("http"));
+
+    }
+    // Initialize API
+    api = new OAIDefaultApi(0);
+    // Setup server config
+    api->setNewServerForAllOperations(
+        url,
+        QString::fromStdString("V2X-Hub Configured PortDrayage WebService"),
+        QMap<QString, OAIServerVariable>()
+    );
+}
+
+    
+OAIContainerActionStatus WebServiceClient::get_current_unloading_action() {
+    return current_unloading_action;
+}
+
+OAIContainerActionStatus WebServiceClient::get_current_loading_action() {
+    return current_loading_action;
+}
+
+OAIInspectionStatus WebServiceClient::get_current_inspection() {
+    return current_inspection;
+}
+
+void WebServiceClient::request_loading_action(std::string vehicle_id, std::string container_id, std::string action_id) {
+    OAIContainerRequest req;
+    QEventLoop loop;
+
+     // Call back for POST /loading/
+    connect(api, &OAIDefaultApi::loadingPostSignal, [&]() {
+        FILE_LOG(logERROR) << "Success /loading POST";
+        loop.quit();
+    });
+    // Error call back for POST /loading/
+    connect(api, &OAIDefaultApi::loadingPostSignalE, [&](QNetworkReply::NetworkError error_code, QString error_str) {
+        FILE_LOG(logERROR) << "Failure /loading POST : " << error_str.toStdString();
+        FILE_LOG(logERROR) << error_code;
+        loop.quit();
+    });
+
+    // Setup request
+    req.setVehicleId( QString::fromStdString( vehicle_id ) ) ;
+    req.setContainerId( QString::fromStdString( container_id ) );
+    req.setActionId( QString::fromStdString( action_id ) );
+
+    FILE_LOG(logERROR) << "Sending loading request : " << req.asJson().toStdString();
+    api->loadingPost( req );
+    QTimer::singleShot(0, &loop, &QEventLoop::quit);
+    loop.exec();
+
+    // Poll loading action until complete
+    pollLoadingAction( req.getActionId() );
```

Comment:
So when the request is sent the first time, the loop inside pollLoadingAction is called. 
If the request is sent again (the second time), before the first loop inside pollLoadingAction ends, is there any impact? Like the requrest needs to wait until the first request and loop finished before process the second request?

Reply:
The same comment applies to unloading and inspection request.

Reply:
will do